modules/smart-agent_kubernetes-common/conf/01-hpa-scale-capacity.yaml

```
@@ -0,0 +1,15 @@
+module: kubernetes
+name: "hpa reached max"
+tip: "hpa reached max for too long, the maximum number of desired Pods has been hit"
+signals:
+  max:
+    metric: kubernetes.hpa.spec.max_replicas
+  current:
+    metric: kubernetes.hpa.status.current_replicas
+  signal:
+    formula: (max-current)
+rules:
+  major:
+    threshold: 0
+    comparator: "="
```

Ok this time it makes sens for me :) however I would like to suggest:

```suggestion
  desired:
    metric: kubernetes.hpa.status.desired_replicas
  signal:
    formula: (max-desired)
rules:
  major:
    threshold: 0
    comparator: ">="
```

I think using `desired` instead of `current` could provide a better result:
- raising alert sooner: as soon as the configuration is pushed (before to wait it has been applied)
- covering all cases: I am afraid if the condition could not be meet so your current will never scale and you will not have any alert whereas it is not able to scale

Notice the correct operator is `==` and not `=` but I also prefer to keep the `>=` which seems more suited here while we want to alert only if current or desired are > to max (and not the reverse).

If you agree you can apply this suggestion directly from github web.

---

I agree with the reasonning, but i will also change the alert description, it should be "hpa is not able to scale up" (imagine a lack of compute resources), the alert is more general and can manage two kinds of issues: lack of ressources, too small hpa

my first intention was only to monitor a full hpa ;)